tests/end-to-end/director/runtime_api_test.go

```diff
@@ -453,6 +450,103 @@ func TestQuerySpecificRuntime(t *testing.T) {
 	assert.Equal(t, createdRuntime.Description, queriedRuntime.Description)
 }
 
+func TestApplicationsForRuntime(t *testing.T) {
+	//GIVEN
+	ctx := context.Background()
+	tenantID := "90b9ccc8-7829-4511-ac17-5b0c872a41b5"
+	tenantApplications := []*graphql.Application{}
+	scenarios := []string{"default", "black-friday-campaign", "christmas-campaign", "summer-campaign"}
+
+	applications := []struct {
+		ApplicationName string
+		Tenant          string
+		WithinTenant    bool
+		Scenarios       []string
+	}{
+		{
+			Tenant:          tenantID,
+			ApplicationName: "noneofscenarios",
+			WithinTenant:    false,
+			Scenarios:       []string{},
+		},
+		{
+			Tenant:          tenantID,
+			ApplicationName: "first",
+			WithinTenant:    true,
+			Scenarios:       scenarios[:1],
```

Comment:
I have to think when I see such line, simplify it to not force me to think too much